include/cantera/oneD/StFlow.h

```
@@ -254,6 +255,87 @@ class StFlow : public Domain1D
+     * points are imposed and thus serve as a boundary condition that affects the
+     * solution of the governing equations in the 1D domain. The imposition of fixed
+     * points in the domain means that the original set of governing equations' boundary
+     * conditions would over-determine the problem. Thus, the boundary conditions are changed
+     * to reflect the fact that the control points are serving as internal boundary conditions.
+     *
+     * The imposition of the two internal boundary conditions requires that two other
+     * boundary conditions be changed. The first is the boundary condition for the continuity equation
+     * at the left boundary, which is changed to be a value that is derived from the solution at the
+     * left boundary. The second is the continuity boundary condition at the right boundary, which is
+     * also determined from the flow solution by using the oxidizer axial velocity equation variable to
+     * compute the mass flux at the right boundary.
+     *
+     * This method is based on the work of M. Nishioka, C.K. Law, and T. Takeno (1996) titled
+     * "A Flame-Controlling Continuation Method for Generating S-Curve Responses with
+     * Detailed Chemistry"
+     **/
+
+    //! Returns the temperature at the left control point
+    double leftControlPointTemperature() const {
+        if (m_twoPointControl && (m_zLeft != Undef)) {
+            return m_tLeft;
+        } else {
+            return NAN;
+        }
+    }
+
+    //! Returns the z-coordinate of the left control point
+    double leftControlPointCoordinate() const {
+        if (m_twoPointControl && (m_zLeft != Undef)) {
+            return m_zLeft;
+        } else {
+            return NAN;
+        }
+    }
+
+    //! Sets the temperature of the left control point
+    void setLeftControlPointTemperature(double temperature) {
+        if (m_twoPointControl && (m_zLeft != Undef)) {
+            m_tLeft = temperature;
+        }
+    }
```

Should setting the control point temperature when two point control isn't enabled, or before the coordinate has been set, be an error? Silently ignoring the input is somewhat surprising behavior.